provider/provider.go

```diff
@@ -143,12 +142,12 @@ func templateCondition(condition []byte, ctx map[string]lib.ChainTemplate) ([]by
 // We need some wrapper that can deal with out of tree providers, this will be a call, that will mock it out, but go against in tree.
 func GetProviderClient(config lib.Config) (Client, error) {
 	switch config.Name {
-	case "go":
-		return golang.NewGolangProvider(config), nil
 	case "java":
 		return java.NewJavaProvider(config), nil
 	case "builtin":
 		return builtin.NewBuiltinProvider(config), nil
+	case "go":
+		return grpc.NewGRPCClient(), nil
```

Comment:
I don't understand how we would break out e.g. the Java provider without this being parameterized in some way

Reply:
Captured in #142 resolving this for now